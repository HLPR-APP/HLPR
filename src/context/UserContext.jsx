import { createContext, useContext, useState } from 'react';
import { getUser } from '../services/users';

const UserContext = createContext();

const UserProvider = ({ children }) => {
  const currentUser = getUser();
  const [user, setUser] = useState(
    currentUser ? { id: currentUser.id, email: currentUser.email } : null
  );
  return (
    <UserContext.Provider value={{ user, setUser }}>
      {children}
    </UserContext.Provider>
  );
};

const useUser = () => {
  const userCtx = useContext(UserContext);
  if (userCtx === undefined) {
    throw new Error('Must be wrapped in a UserProvider to use useUser');
  }
  return userCtx;
};

export { UserProvider, useUser };
