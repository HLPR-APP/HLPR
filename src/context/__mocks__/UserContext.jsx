import { createContext, useContext, useState } from 'react';

const UserContext = createContext();

const UserProvider = ({ mockUser, children }) => {
  const [user, setUser] = useState(
    mockUser ? { id: mockUser.id, email: mockUser.email } : null
  );
  return (
    <UserContext.Provider value={{ user, setUser }}>
      {children}
    </UserContext.Provider>
  );
};

const useUser = () => {
  const userCtx = useContext(UserContext);
  if (userCtx === undefined) {
    throw new Error('Must be wrapped in a UserProvider to use useUser');
  }
  return userCtx;
};

export { UserProvider, useUser };
