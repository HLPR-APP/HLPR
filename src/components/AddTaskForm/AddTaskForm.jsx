import {
  Button,
  FormControl,
  FormLabel,
  Input,
  Checkbox,
  Flex,
  Heading,
  Link,
  Stack,
  Image,
} from '@chakra-ui/react';
import { useState } from 'react';
import { useUser } from '../../context/UserContext';
import { AddTask } from '../../services/services';

export default function AddTaskForm() {
  const auth = useUser();
  const [taskname, setTaskName] = useState('');
  const [description, setDescription] = useState('');
  const [imageURL, setImageURL] = useState('');

  const handleSubmit = async (e) => {
    e.preventDefault();
    const response = await AddTask(
      auth.user.email,
      taskname,
      description,
      imageURL
    );
    console.log(response);
    window.location.reload();
  };

  return (
    // <form onSubmit={handleSubmit}>
    //   <FormControl m="10">
    //     <FormLabel htmlFor="taskname">Task Name</FormLabel>
    //     <Input
    //       id="taskname"
    //       value={taskname}
    //       onChange={(e) => setTaskName(e.target.value)}
    //     />
    //     <FormLabel htmlFor="description">Description</FormLabel>
    //     <Input
    //       id="description"
    //       value={description}
    //       onChange={(e) => setDescription(e.target.value)}
    //     />
    //     <FormLabel htmlFor="date">Image-URL</FormLabel>
    //     <Input
    //       id="imageURL"
    //       value={imageURL}
    //       onChange={(e) => setImageURL(e.target.value)}
    //     />
    //   </FormControl>
    //   <Button type="submit">Submit Task</Button>
    // </form>

    <Stack minH={'40vh'} direction={{ base: 'column', md: 'row' }}>
      <Flex p={8} flex={1} align={'center'} justify={'center'}>
        <Stack spacing={4} w={'full'} maxW={'md'}>
          <Heading fontSize={'2xl'}>Create A Task</Heading>
          <form onSubmit={handleSubmit}>
            <FormControl>
              <FormLabel htmlFor="taskname">Task Name</FormLabel>
              <Input
                id="taskname"
                value={taskname}
                onChange={(e) => setTaskName(e.target.value)}
              />
            </FormControl>
            <FormControl>
              <FormLabel htmlFor="description">Description</FormLabel>
              <Input
                id="description"
                value={description}
                onChange={(e) => setDescription(e.target.value)}
              />
            </FormControl>
            <FormControl>
              <FormLabel htmlFor="date">Image-URL</FormLabel>
              <Input
                id="imageURL"
                value={imageURL}
                onChange={(e) => setImageURL(e.target.value)}
              />
            </FormControl>
            <Stack spacing={6}>
              <Stack
                direction={{ base: 'column', sm: 'row' }}
                align={'start'}
                justify={'space-between'}
              >
                {/* <Checkbox>Remember me</Checkbox>
                <Link color={'blue.500'}>Forgot password?</Link> */}
              </Stack>
              <Button type="submit" colorScheme={'blue'} variant={'solid'}>
                Create Task
              </Button>
            </Stack>
          </form>
        </Stack>
      </Flex>
      <Flex flex={1}>
        <Image
          pr="10"
          pt="5"
          alt={'Login Image'}
          objectFit={'cover'}
          src={
            'https://images.unsplash.com/photo-1484480974693-6ca0a78fb36b?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1472&q=80'
          }
        />
      </Flex>
    </Stack>
  );
}
