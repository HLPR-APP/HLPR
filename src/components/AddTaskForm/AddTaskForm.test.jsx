import dotenv from 'dotenv';
dotenv.config();
import React from 'react';
import { render } from '@testing-library/react';
import { UserProvider } from '../../context/UserContext.jsx';
import AddTaskForm from './AddTaskForm.jsx';

it('should render form', () => {
  const { container } = render(
    <UserProvider>
      <AddTaskForm />
    </UserProvider>
  );
  expect(container).toMatchSnapshot();
});
