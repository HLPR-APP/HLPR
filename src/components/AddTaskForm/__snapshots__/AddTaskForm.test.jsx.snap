// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render form 1`] = `
<div>
  <div
    class="chakra-stack css-1qghqig"
  >
    <div
      class="css-45tizf"
    >
      <div
        class="chakra-stack css-1bpi8v6"
      >
        <h2
          class="chakra-heading css-1n49pl7"
        >
          Create A Task
        </h2>
        <form>
          <div
            class="chakra-form-control css-0"
            role="group"
          >
            <label
              class="chakra-form__label css-uvho5g"
              for="taskname"
              id="field-1-label"
            >
              Task Name
            </label>
            <input
              class="chakra-input css-0"
              id="taskname"
              value=""
            />
          </div>
          <div
            class="chakra-form-control css-0"
            role="group"
          >
            <label
              class="chakra-form__label css-uvho5g"
              for="description"
              id="field-2-label"
            >
              Description
            </label>
            <input
              class="chakra-input css-0"
              id="description"
              value=""
            />
          </div>
          <div
            class="chakra-form-control css-0"
            role="group"
          >
            <label
              class="chakra-form__label css-uvho5g"
              for="date"
              id="field-3-label"
            >
              Image-URL
            </label>
            <input
              class="chakra-input css-0"
              id="imageURL"
              value=""
            />
          </div>
          <div
            class="chakra-stack css-16voevi"
          >
            <div
              class="chakra-stack css-1bnf7oa"
            />
            <button
              class="chakra-button css-5c2176"
              type="submit"
            >
              Create Task
            </button>
          </div>
        </form>
      </div>
    </div>
    <div
      class="css-13pmxen"
    >
      <img
        alt="Login Image"
        class="chakra-image css-17n63z0"
        src="https://images.unsplash.com/photo-1484480974693-6ca0a78fb36b?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1472&q=80"
      />
    </div>
  </div>
</div>
`;
